code/datums/status_effects/debuffs/heart_attack.dm

```diff
@@ -0,0 +1,164 @@
+			owner.adjustStaminaLoss(20)
+			owner.emote("cough")
+			oxyloss_sum += 8
+
+	if(time_until_stoppage <= ATTACK_STAGE_FOUR) //And now we compound it with even worse effects.
+		owner.stop_sound_channel(CHANNEL_HEARTBEAT)
+		sound = FALSE
+		owner.playsound_local(owner, 'sound/effects/singlebeat.ogg', 100, FALSE, use_reverb = FALSE)
+
+		if(SPT_PROB(5, seconds_between_ticks))
+			to_chat(owner, span_userdanger("It feels like you're shutting down..."))
+			owner.adjust_dizzy_up_to(4 SECONDS, 10 SECONDS)
+			owner.adjust_eye_blur_up_to(4 SECONDS, 20 SECONDS)
+			owner.adjustStaminaLoss(20)
+
+		if(SPT_PROB(5, seconds_between_ticks))
+			owner.emote("cough")
+			if(SPT_PROB(5, seconds_between_ticks))
+				to_chat(owner, span_userdanger("You cough. Everything goes dark. You're going to die soon."))
+				owner.adjust_temp_blindness(10 SECONDS) //Are you panicking yet? You should be panicking by now.
+			else
+				to_chat(owner, span_userdanger("As you cough, your chest surges in pain and darkness closes in around your sight."))
+				owner.adjust_temp_blindness(2 SECONDS)
+				owner.adjust_eye_blur_up_to(4 SECONDS, 20 SECONDS)
+			oxyloss_sum += 8
+			owner.Paralyze(1 SECONDS)
+		oxyloss_sum += 3
+
+	if(owner.getOxyLoss() < OXYLOSS_MAXIMUM) //A bad enough roll on the verge of passing out might still push you over into unconciousness for a few seconds...?
+		owner.adjustOxyLoss(oxyloss_sum)
+
+	if(time_until_stoppage <= 0)
+		if(owner.stat == CONSCIOUS)
+			to_chat(owner, span_userdanger("You feel a terrible pain in your chest, as if your heart has stopped!"))
+		owner.adjust_eye_blur(20 SECONDS)
+		human_owner.set_heartattack(TRUE)
+		owner.reagents.add_reagent(/datum/reagent/medicine/c2/penthrite/heart_attack, 2) // To give the victim a final chance to shock their heart before losing consciousness
+		var/flash_type = /atom/movable/screen/fullscreen/flash
+		if(owner.client?.prefs?.read_preference(/datum/preference/toggle/darkened_flash))
+			flash_type = /atom/movable/screen/fullscreen/flash/black
+		owner.overlay_fullscreen("flash", flash_type)
+		addtimer(CALLBACK(owner, TYPE_PROC_REF(/mob/living, clear_fullscreen), "flash", 1 SECONDS), 1 SECONDS)
+		qdel(src)
+		return FALSE
+
+	time_until_stoppage--
+
+/datum/status_effect/heart_attack/get_examine_text()
+	if(time_until_stoppage <= ATTACK_STAGE_THREE)
+		var/mob/living/carbon/human/human_owner = owner
+		var/cannot_grasp = (human_owner.usable_hands <= 0 || owner.incapacitated & INCAPABLE_RESTRAINTS || HAS_TRAIT(owner, TRAIT_HANDS_BLOCKED))
+		return span_warning("[owner.p_they()] looks to be doubling over" + "[cannot_grasp ? " in pain!" : ", clutching [owner.p_their()] chest in pain!"]")
+
+/datum/status_effect/heart_attack/on_remove()
+	REMOVE_TRAIT(owner, TRAIT_DISEASELIKE_SEVERITY_HIGH, type)
+	owner.med_hud_set_status()
+	. = ..()
+
+///End the heart attack due to heart removal.
+/datum/status_effect/heart_attack/proc/end_attack(datum/source, obj/item/organ)
```

Comment:
This proc name isn't really accurate, should probably just be `on_organ_removed`